library/std/src/sys/uefi/pipe.rs

```diff
@@ -0,0 +1,188 @@
+//! An implementation of Pipes using UEFI variables
```

Comment:
Perhaps it would be a better approach to temporarily overwrite the console handles in the system table with custom implementations, so that e.g. drivers not aware of this protocol work properly?

Reply:
I did try that quite a while ago; changing anything in the SystemTable pointer of the child handle seemed to blow up everything in my face. But maybe there is a safe way to change the console handles I don't know about?